### ISSUE TITLE
UNUSED_LOCAL_VARIABLE: False positive: cfloop index

### ISSUE THREAD

**Issue by jaredbeck** [cflint/CFLint#693]
```cfml
<cfquery ..redacted..>
  <cfloop list="#variables.redacted#" index="local.tuple"> <!--- line 104 --->
    #Val(ListGetAt(tuple, 1, '|'))#
```

```
Message code:UNUSED_LOCAL_VARIABLE
        File:redacted.cfc
        Column:25
        Line:104
                Message:Local variable tuple is not used in function getResults. Consider removing it.
                Variable:'tuple' in function: getResults
                Expression:tuple
```

Possibly related to https://github.com/cflint/CFLint/issues/657

**Comment by jaredbeck** [comment]
A more complete example:

```cfml
<cffunction name="f">
  <cfquery>
    <cfloop list="a,b,c" index="local.x">
      #x#
    </cfloop>
  </cfquery>
</cffunction>
```

```
Message code:UNUSED_LOCAL_VARIABLE
	File:/Users/jared/git/pf/tafkan/webroot/tmp/lint_ulv.cfm
	Column:11
	Line:3
		Message:Local variable x is not used in function f. Consider removing it.
		Variable:'x' in function: f
		Expression:x
```

**Comment by jaredbeck** [comment]
For comparison, the following passes:

```cfml
    <cfset var x = 0>
    <cfloop list="a,b,c" index="x">
```

**Comment by jaredbeck** [comment]
To further simplify the example:

```cfml
<cffunction name="f">
  <cfset var x = 0>
  <cfquery datasource="#request.app_datasource#">
    #x#
  </cfquery>
</cffunction>
```

It seems that using a variable in `cfquery` is not recognized.

**Comment by jaredbeck** [comment]
This issue is very similar to https://github.com/cflint/CFLint/issues/632. If it's helpful to you to combine them, @ryaneberly, please feel free to.

**Comment by CreativeNotice** [comment]
Can confirm this happens when in script syntax as well. In this example, `header` will trigger the rule.
```coldfusion
for( var header in requestHeaders ){
	cfhttpparam( type = 'header', name = header.name, value = header.value );
}
```
